### DISCUSSION TITLE
Uncaught Snapshot missing on Livewire component with id: 41SHDhywQvJfvmouU5eh

### DISCUSSION THREAD

**Discussion by azamspeare** [livewire/livewire#6698] [7. Bugs (archived)] [answered]
### Livewire version

3

### Which PHP version are you using?

PHP 8.1

### Repository URL

https://github.com/Abduazam/laravel-template

### Steps To Reproduce

I am using Livewire3 for making crud for a specific model (Languages). In blade I am showing data in table and it works correctly in first render. When I search it only can find first row's of table's data, but when search others it gives an error: Uncaught Snapshot missing on Livewire component with id: specific id.

Livewire's Index component:
```
class Index extends Component
{
    use SearchTrait;

    public function render(LanguageRepository $repository): View
    {
        return view('livewire.features.languages.index', [
            'languages' => $repository->getFiltered($this->search)
        ]);
    }
}
```

Query:
```
public function getFiltered($search)
    {
        return Language::select('id', 'slug', 'title', 'thumbnail', 'created_at')
            ->when($search, function ($query, $search) {
                return $query->where('title', 'like', '%' . $search . '%');
            })->get();
    }
```

Index blade:
```
<div class="block block-rounded">
    <div class="block-content">
        <div class="filter-table pb-4">
            <div class="row w-100 h-100 m-0 p-0">
                <div class="col-lg-4 p-0">
                    <label for="search" class="w-100 ps-0">
                        <input wire:model.live.debounce.200ms="search" type="text" class="form-control form-control-sm w-100"
                               id="search" name="search" placeholder="Search" >
                    </label>
                </div>
            </div>
        </div>
        <ul>
            <table class="table-bordered">
                @foreach($languages as $item)
                    <tr>
                        <td class="px-2 py-1">{{ $item->title }}</td>
                        <td class="px-2 py-1">@livewire('features.languages.edit', [$item], key("language-{$item->id}"))</td>
                    </tr>
                @endforeach
            </table>
        </ul>
    </div>
</div>
```

Edit component:
```
class Edit extends Component
{
    public Language $item;

    public function mount($item): void
    {
        $this->item = $item;
    }

    public function render(): View
    {
        return view('livewire.features.languages.edit');
    }
}
```

Edit blade:
```
<div>
    {{ $item->id }}
</div>
```

Result:
![image](https://github.com/livewire/livewire/assets/58428722/e5c5f329-43c8-44b7-9566-fa097b5271a5)

Search case:
https://github.com/livewire/livewire/assets/58428722/05770bd8-1ac8-47d9-a8eb-08faf734d7a6

How can I solve this issue? I have been facing this problem for several days. Still haven't found a solution

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@Abduazam please don't paste screenshots of code, please paste the code in code blocks.

When you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

**Reply by azamspeare** [reply]
Thanks, edited!

**Reply by joshhanley** [reply, marked as answer]
@Abduazam thanks! Can you try also adding a key on the `<tr>` element, like `<tr wire:key="language-row-{{$item->id}}">`?

**Reply by azamspeare** [reply]
Oh man thank you! Worked

**Comment by swaneswan** [comment]
I believe this is related to #6704. I am also having this issue. Have you found any solutions?

**Reply by joshhanley** [reply]
@swaneswan while the error is the same, I don't think it's related as @Abduazam is saying this is happening locally where that issue is related to Cloudflare

**Reply by azamspeare** [reply]
Yes, it worked. I have marked the answer

**Comment by DerekBuntin** [comment]
I'm having this very same issue but the answers above haven't helped, this worked fine in Livewire 2.

`Uncaught Snapshot missing on Livewire component with id: kyWnEeRYHQrQsMHazax5`

I have a full-page component which lists articles in a table which is autogenerated using a Laravel dynamic component to generate the table columns.

One of those columns is a nested component for toggling the published column, a boolean toggle, which works fine.

I also created a draggable <tr> inside a foreach loop with a wire:key added, this <tr> element also has other values for dragging the row and a class.

When I drag one of the rows to reorder the list the component is called and the reordering works but the error above is shown on the error console and the child toggle component isn't loaded, the ID in the above error relates to that child toggle component.

I also have dropdown filters where I can select to filter by category, which works sometimes but others not with the following errors:

`
Uncaught Snapshot missing on Livewire component with id: FdOa66hIr5tH5ZyqUCiL
Uncaught (in promise) Component not found: FdOa66hIr5tH5ZyqUCiL
`

The search feature on this full-page component works fine and returns the correct rows with no error, it only seems to be after the component call using the draggable script and when filtering.

I've read the docs in detail and tried almost everything but to no avail, any help would be appreciated.

When I remove the nested component there are no issues, I'll take another look at it with fresh eyes and go over the docs again in case I've missed something in the upgrade.

**Reply by azamspeare** [reply]
Hi @DerekBuntin 
Can you share code to review it? Because it helps to understand more and get answer asap

**Reply by DerekBuntin** [reply]
Hi @Abduazam,

I removed the child toggle component and the reordering works without issue so there must be something going on with the child component, although on the first load it works fine too.

I'll review the docs again and will update you with the code and hopefully with a resolution shortly.

**Reply by DerekBuntin** [reply]
Hi @Abduazam,

Here goes, I've developed a CMS and created packages to keep the code organised, the namespace is therefore a custom package.

Here is the ToggleButton Livewire Component:

```
namespace Adonis\Cms\Http\Livewire;

use Illuminate\Database\Eloquent\Model;
use Livewire\Component;

class ToggleButton extends Component
{
    
   public Model $model;

    public string $field;

    public bool $isActive;

    public function mount()
    {
        $this->isActive = (bool) $this->model->getAttribute($this->field);
    }

    public function updating($field, $value): void
    {
        $this->model->setAttribute($this->field, $value)->save();
    }

    public function render()
    {
        return view('cms::livewire.toggle-button', ['id' => $this->model->id, 'field' => $this->field]);
    }
}
```

Here is the blade file for the component:

```
<div x-data="{ value: @entangle('isActive').live }"
    class="flex items-center justify-center"
    x-id="['toggle-label']"
    x-cloak>

    <label for="{{ $id }}-{{ $field }}" class="sr-only">{{ __($field) }}</label>
    <input wire:model.live="isActive" type="hidden" name="{{ $field }}" id="{{ $id }}-{{ $field }}" :value="value ? '1' : '0'" />

    <button
        @click="value = ! value"
        type="button"
        role="switch"
        :aria-checked="value"
        :aria-labelledby="$id('toggle-label')"
        :class="value ? 'bg-green-500' : 'bg-red-500'"
        class="relative inline-flex w-11 h-6 rounded-full transition border-2 border-transparent"
    >
        <span
            :class="value ? 'translate-x-5' : 'translate-x-0'"
            class="bg-white h-5 w-5 rounded-full transition shadow-md"
            aria-hidden="true"
        >
            <span
                :class="value ? 'opacity-0 ease-out duration-100' : 'opacity-100 ease-in duration-200'"
                class="absolute inset-0 h-full w-full flex items-center justify-center transition-opacity"
                aria-hidden="true">
              <svg class="h-3 w-3 text-red-500" fill="none" viewBox="0 0 12 12">
                <path d="M4 8l2-2m0 0l2-2M6 6L4 4m2 2l2 2" stroke="currentColor" stroke-width="2" stroke-linecap="round"
                      stroke-linejoin="round"/>
              </svg>
            </span>
            <span
                :class="value ? 'opacity-100 ease-in duration-200' : 'opacity-0 ease-out duration-100'"
                class="absolute inset-0 h-full w-full flex items-center justify-center transition-opacity"
                aria-hidden="true">
              <svg class="h-3 w-3 text-green-500" fill="currentColor" viewBox="0 0 12 12">
                <path
                    d="M3.707 5.293a1 1 0 00-1.414 1.414l1.414-1.414zM5 8l-.707.707a1 1 0 001.414 0L5 8zm4.707-3.293a1 1 0 00-1.414-1.414l1.414 1.414zm-7.414 2l2 2 1.414-1.414-2-2-1.414 1.414zm3.414 2l4-4-1.414-1.414-4 4 1.414 1.414z"/>
              </svg>
            </span>
        </span>
    </button>
</div>
```

I have tried to add another empty div around the code in this blade file to no avail. This component works fine on first load just not when there is a change to the parent component.

Here is how it is called within a dynamic blade file represented as a column <td> within a table row <tr> that has a wire:key assigned to the row:

```
@props(['value','model','key','order'])

<td class="px-6 py-4 whitespace-no-wrap text-center text-sm leading-5 text-gray-500">
    <livewire:toggle-button :model="$model" field="published" :key="$key" />
</td>
```

The following code is the table row, you'll notice the loop calling the dynamic component and passing the properties over, this all works fine:

```
@if(!isset($row->parent_id) || isset($row->parent_id) && $row->parent_id === 0)
    <tr id="item-{{ $row->id }}"
        wire:key="row-{{ $row->id }}"
        @if(isset($use_ordering) && $use_ordering && isset($row->order))
        drag-item="{{ $row->id }}"
        drag-item-order="{{ $row->order }}"
        @if(isset($menu))
            drag-item-menu="{{ $menu->id }}"
        @endif
        @if(isset($category))
        drag-item-category="{{ $category }}"
        @endif
        draggable="true"
        @endif
        class="@if(method_exists($row, 'isScheduled') && $row->isScheduled() && $row->isScheduled() && isset($row->published)) odd:border-l odd:border-yellow-500 odd:border-l-4 even:border-l even:border-yellow-500 even:border-l-4 @elseif(isset($row->published) && $row->published) odd:border-l odd:border-green-500 odd:border-l-4 even:border-l even:border-green-500 even:border-l-4 @elseif(isset($row->published) && !$row->published) odd:border-l odd:border-red-500 odd:border-l-4 even:border-l even:border-red-500 even:border-l-4 @endif odd:bg-white even:bg-gray-50 cursor-pointer">
        @foreach($this->columns() as $column)
            <x-dynamic-component
                :component="$column->component"
                :value="$row[$column->key]"
                :column="$column->key"
                :model="$row"
                :key="'row-'.$row->order"
                :route="$route"
                :countries="$countries ?? ''"
                :filters="$filters"
            >
            </x-dynamic-component>
        @endforeach
    </tr>

    @if(isset($row->children) && $row->children->count() > 0)
        @foreach($row->children as $child)
            @include('cms::admin.table.child', ['item' => $child, 'depth' => $depth+1])
        @endforeach
    @endif
@endif
```


The following code is the table that resides within the full-page component layout:

```
<table class="min-w-full">
                    <thead>
                    <tr>
                        @foreach($this->columns() as $column)
                            <x-dynamic-component
                                :component="$column->header"
                                :column="$column"
                                :route="$route"
                                :sortBy="$sortBy"
                                :sortDirection="$sortDirection"
                                :key="$column->key"
                                :filters="$filters"
                            >
                            </x-dynamic-component>
                        @endforeach
                    </tr>
                    </thead>
                    <tbody @if(isset($use_ordering) && $use_ordering) drag-root="reorder" @endif>
                        @foreach($rows as $row)
                            @includeIf('cms::admin.table.parent', ['row'=> $row, 'depth' => 0])
                        @endforeach
                    </tbody>
                </table>
```
                
I switched off the child toggle component to try the reordering and it works fine without issue so there must be an issue with how the child component is being called or rendered in some way.

This worked perfectly fine in LW 2, I just upgraded yesterday.

I hope this makes sense, clearly, there is something that's changed from V2 to V3 that I seem to be missing.
